libpdf/models/horizontal_box.py

```diff
@@ -26,13 +32,19 @@ def __init__(
         y0: float = None,
         x1: float = None,
         y1: float = None,
+        ncolor: tuple = None,
+        fontname: str = None,
     ):
         """Init the class with plain char of a character and its rectangular coordinates."""
         self.x0 = x0
         self.y0 = y0
         self.x1 = x1
         self.y1 = y1
         self.text = text
+        self.ncolor = ncolor
+        self.fontname = fontname
+        self.bold = True if "Bold" in fontname else False
+        self.italic = True if "Italic" in fontname else False
```

Comment:
This logic *should* not break, but I'm not sure about that. I've seen fonts that are exposed like this
![image](https://github.com/useblocks/libpdf/assets/839934/2e7542e9-48b8-41bb-9f3e-ff2ec0c9e03c)
Again I think this logic can be done by an end user.

Reply:
Ok, your are the pdf export. Exposing the fontname should follow the same strategy like for ncolor? store fontname on each char, return fontname for word if all chars in the word have same font and so on.

Reply:
I just checked fonts installed in my system. And as PDF can embed fonts, my logical conclusion is that the logic may fail. But having the font name is a really good idea. Also font size would be good. Yes, I'd propagate it as well as you did for ncolor.